### PR TITLE
add resource requests to sidecars and e2e test to prevent regression

### PR DESCRIPTION
**Pull request by sjenning**
**What this PR does / why we need it**:
Follow up to https://github.com/openshift/hypershift/pull/1051 to make sure none of the HCP pods run as `BestEffort` and we have scheduler accounting for ever container in the HCP.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Review by alvaroaleman — APPROVED** [review]

File: test/e2e/util/util.go [hunk 1]
```diff
@@ -435,3 +435,27 @@ func EnsureAPIBudget(t *testing.T, ctx context.Context, client crclient.Client,
 		}
 	})
 }
+
+func EnsureHCPContainersHaveResourceRequests(t *testing.T, ctx context.Context, client crclient.Client, hostedCluster *hyperv1.HostedCluster) {
```
Comment (csrwng) [inline test/e2e/util/util.go:439]: nice

**Review by csrwng — APPROVED** [review]

File: test/e2e/api_budget_test.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import (
+	"context"
+	"testing"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestAPIBudget(t *testing.T) {
```
Comment (ironcladlou) [inline test/e2e/api_budget_test.go:15]: What's the purpose of this one?

/hold

Comment (sjenning) [reply test/e2e/api_budget_test.go:15]: gah, how did that get included! :confounded:

**Review by alvaroaleman — APPROVED** [review]
/hold cancel

**Pull request merged**